### PR TITLE
v0.0.9 - Circuit Audit Fixes

### PR DESCRIPTION
**Pull request by richardliang**
### Description

For circuit updates, please provide a brief description of the changes included in this PR.
* Fix index shift / custom message malleability attack with new actor and payee regexes. details in this PR https://github.com/zkp2p/zk-p2p/pull/119/commits/ef083f237c693db81bc6782d5bbadad779ce7b90
* Optimized regexes: timestamp, from, body hash regex from zkemail v2.0.1
* Remove unnecessary circuits https://github.com/zkp2p/zk-p2p/pull/119/commits/30067df38607ea41f27b798a45fe6e7da7557b95
* Added comma extraction and created new venmo_send_amount regex. This is a breaking change for SC and UI. Need to make amount public signals length 2 vs length 1 previously.
* Copying over Sora's implementation https://github.com/zkemail/zk-email-verify/pull/124/files
* Fix typos https://github.com/zkp2p/zk-p2p/pull/111
* Note: need to update README and other docs

### Checklist

For any circuit changes, complete the following steps to ensure faster integration with UI and Smart Contracts:

- [x] Ran `yarn:compile:XXX` where `XXX` is the name of the circuit
- [x] Ran `yarn test` to check all tests are passing
- [x] Ran `yarn:genkey:chunked:unsafe:XXX` AND `yarn:genkey:non-chunked:unsafe:XXX` to generate new proving keys for each circuit that was updated
- [x] Updated `UPLOAD_FOLDER` in `circuit.env` to a new path in S3. E.g. `UPLOAD_FOLDER="v2/v0.0.5"`
- [x] Ran `yarn uploadkeys:XXX` to upload to S3. Ensure that BOTH chunk and nonchunk keys are uploaded
- [x] Ran `yarn:genverifier:XXX` to generate solidity verifier
- [x] Ran `yarn:gen-input:XXX` to generate inputs for witness generation
- [x] Ran `yarn:genwitness:XXX` to generate witnesses for proof generation
- [x] Ran `yarn:genproof:XXX` to generate proof for test solidity calldata
- [x] Ran `yarn:gencalldata:XXX` to generate solidity calldata
- [x] Ensured files are public through ACL in `zk-p2p` S3 bucket
- [x] Updated `package.json` in circuits to correct version and ran `npm publish`
- [x] Paste test calldata below

send
["0x226c7510f1bc5e74f5ad3bd370e3f0fccb9ea2125b79be9ba4fcb7a266c438e7", "0x06019100116fb49a5c846963d6b3342bf0c41aaa3e1c70f21add1649cc47ea69"],[["0x2716a896339b0cd0b12c1860e8cf1c647cbde473216f2b3d4a5e2f8930f5b65b", "0x0625204290ee08b13b2a6ac210c880eeca7192952137734ef9e8927d05c17d7c"],["0x1068f5cccca7f9b428df8dcbf8180e5aa597cdb4e40130b7208057ccf23611d9", "0x07793b9fb030bfb0aa5d3598e3dcdbe9d31698045404f434ba573966eef21bb1"]],["0x0afd380d08eac5415ce73074230162f7efdedf550b548ca3ab0cdd92024d8122", "0x10449a83449773b49035bbae972a5688a31384b37ba326f345e4afc1c6eeee13"],["0x2cf6a95f35c0d2b6160f07626e9737449a53d173d65d1683263892555b448d8f","0x0000000000000000000000000000000000000000000000000076406f6d6e6576","0x000000000000000000000000000000000000000000000000006f632e6f6d6e65","0x000000000000000000000000000000000000000000000000000000000000006d","0x00000000000000000000000000000000000000000000000000000030302e3033","0x0000000000000000000000000000000000000000000000000000000000000000","0x0000000000000000000000000000000000000000000000000033373633383631","0x0000000000000000000000000000000000000000000000000000000000393334","0x14465b905804512ffd4ed3425fd0d9d0f50947bddfd35bb55be9c29cca1e1c3a","0x1e1d37257be43685d944597f63d00a6b638b9b4ff1752cf8db9c5de48ececbe8","0x0000000000000000000000000000000000000000000000000000000000003039"]

registration
["0x2371ed99cdec3d982e7d98e7fd59c520a37e7b6eea1edc7827f4a997a5514e3e", "0x0b5bfba2051eb8fbf9ba53f7680df919ac1aaafa39e5a97568a0240db05ef0a8"],[["0x2d88b926044ace956f76de521cfe0eae56d6daa7b68a7a928ce4806ef5fbcd94", "0x0c5435223cd810eaddeec5e443e6165193de43074828c36bcbe580c9c8320adc"],["0x228cd238596bd97ffdaf2c5ff7ea2758deb5fd7816e27b3488b3174d4033e8ec", "0x05d09883ebadc49de90a960bba69a8c0774ba90f50ff5882886962c4084c3ec1"]],["0x23259260d22a921ab46eecb5992e1c1941fd4519f4ed871a0e4dc1759f65193b", "0x296c9072575fd76e803c2ea2cffa1df2320cf8453d38a2aa25fa566ded6120fe"],["0x2cf6a95f35c0d2b6160f07626e9737449a53d173d65d1683263892555b448d8f","0x0000000000000000000000000000000000000000000000000076406f6d6e6576","0x000000000000000000000000000000000000000000000000006f632e6f6d6e65","0x000000000000000000000000000000000000000000000000000000000000006d","0x0741728e3aae72eda484e8ccbf00f843c38eae9c399b9bd7fb2b5ee7a055b6bf"]

### CODE REVIEW

File: circuits-circom/venmo_send.circom [hunk 1]
```diff
@@ -15,11 +15,11 @@ template VenmoSendEmail(max_header_bytes, max_body_bytes, n, k, pack_size) {
     assert(n * k > 1024); // constraints for 1024 bit RSA
 
     // Rounded to the nearest multiple of pack_size for extra room in case of change of constants
-    var max_email_amount_len = 7; // Allowing max 4 fig amount + one decimal point + 2 decimal places
+    var max_email_amount_len = 8; // Allowing max 5 fig amount + one decimal point + 2 decimal places. e.g. $2,500.00
```
Comment (0xSachinK) [inline circuits-circom/venmo_send.circom:18]:
```javascript
var max_email_amount_len = 8; // Allowing max 4 fig amount + one comma + one decimal point + 2 decimal places. Max allowed amount $9,999.00 is above the current Venmo limit of $4,999.00
```


The maximum amount that can be sent via Venmo to another user in a P2P payment is $4,999.99 per transaction, with the same amount as the maximum weekly limit for such payments. [Source](https://wise.com/us/blog/venmo-transfer-limit#:~:text=Please%20remember%20that%20for%20person,Take%20a%20look%20at%20Wise)

File: circuits-circom/regexes/venmo_timestamp.circom
Comment (0xSachinK) [inline circuits-circom/regexes/venmo_timestamp.circom:1]: Validated regex optimization.

File: circuits-circom/regexes/venmo_payer_id.circom
Comment (0xSachinK) [inline circuits-circom/regexes/venmo_payer_id.circom:1]: Validated regex optimization.

File: circuits-circom/regexes/venmo_payee_id.circom
Comment (0xSachinK) [inline circuits-circom/regexes/venmo_payee_id.circom:1]: Validated regex optimization.

File: circuits-circom/regexes/venmo_amount.circom
Comment (0xSachinK) [inline circuits-circom/regexes/venmo_amount.circom:1]: Validated regex optimization.

File: circuits-circom/regexes/venmo_actor_id.circom
Comment (0xSachinK) [inline circuits-circom/regexes/venmo_actor_id.circom:1]: Validated regex optimization.

**Review by 0xSachinK — CHANGES_REQUESTED** [review]
The changes look good to me. Nothing that we saved ~200K, ~150K and ~100K constraints on send, receive, and registration circuits, respectively.

File: circuits-circom/venmo_send.circom (hunk 1 above)
Comment (richardliang) [reply circuits-circom/venmo_send.circom:18]: Fixed https://github.com/zkp2p/zk-p2p/pull/131/files

**Comment by kcharbo3** [comment]
LGTM!

**Pull request merged**